config/gcos.words

```diff
@@ -0,0 +1,470 @@
+# GnuCOBOL compiler - list of reserved words
+#
+# Copyright (C) 2016-2017,2021-2022 Free Software Foundation, Inc.
+# Written by Simon Sobisch, Edward Hart, David Declerck, Fabrice Le
+# Fessant, Nicolas Berthier
+#
+# This file is part of GnuCOBOL.
+#
+# The GnuCOBOL compiler is free software: you can redistribute it
+# and/or modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# GnuCOBOL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with GnuCOBOL.  If not, see <https://www.gnu.org/licenses/>.
+
+# Word list for dialect: GCOS 7 (Bull)
+
+# NOTE: the mapping for COMP* usages does not exactly match the GCOS
+# specification, due to the former missing some of the representations
+# required.  The mapping below associates each COMP* with the closest
+# representation currently available.
+
+reserved:	ACCEPT
+reserved:	ACCESS
+reserved:	ACTUAL
+reserved:	ADD
+reserved:	ADDRESS
+reserved:	ADVANCING
+reserved:	AFTER
+reserved:	ALL
+reserved:	ALPHABET
+reserved:	ALPHABETIC
+reserved:	ALPHABETIC-LOWER
+reserved:	ALPHABETIC-UPPER
+reserved:	ALPHANUMERIC
+reserved:	ALPHANUMERIC-EDITED
+reserved:	ALSO
+reserved:	ALTER
+reserved:	ALTERNATE
+reserved:	AND
+reserved:	ANSI
+reserved:	ANY
+reserved:	APPLY			# not supported yet
+reserved:	ARE
+reserved:	AREA
+reserved:	AREAS
+reserved:	ASA
+reserved:	ASCENDING
+reserved:	ASCII
+reserved:	ASSIGN
+reserved:	AT
+reserved:	AUTHOR
+reserved:	B-AND
+reserved:	B-NOT
+reserved:	B-OR
+reserved:	BEFORE
+reserved:	BINARY
+reserved:	BIT
+reserved:	BLANK
+reserved:	BLOCK
+reserved:	BOOLEAN
+reserved:	BOTTOM
+reserved:	BSN
+reserved:	BY
+reserved:	CALL
+reserved:	CANCEL
+reserved:	CD
+reserved:	CF
+reserved:	CH
+reserved:	CHARACTER
+reserved:	CHARACTERS
+reserved:	CLASS
+reserved:	CLOCK-UNITS
+reserved:	CLOSE
+reserved:	COBOL
+reserved:	CODE
+reserved:	CODE-SET
+reserved:	COLLATING
+reserved:	COLUMN
+reserved:	COMMA
+reserved:	COMMON
+reserved:	COMMUNICATION
+reserved:	COMP=COMP-3
+reserved:	COMP-1=BINARY-SHORT
+reserved:	COMP-2=BINARY-LONG
+reserved:	COMP-3
+reserved:	COMP-5=PACKED-DECIMAL
+reserved:	COMP-8=PACKED-DECIMAL
+reserved:	COMP-9			# single precision
+reserved:	COMP-10			# double precision
+reserved:	COMP-15			# quadruple precision
+reserved:	COMPUTATIONAL=COMP-3
+reserved:	COMPUTATIONAL-1=BINARY-SHORT
+reserved:	COMPUTATIONAL-2=BINARY-LONG
+reserved:	COMPUTATIONAL-3
```

Comment:
Doesn't this need the `=COMP-3`, too?

Reply:
In `reserved.c`, I found `COMP-3` and `COMPUTATIONAL-3` are both defined as `COMP_3` already. And with the current words list `cobc -std=gcos-strict --list-reserved` gives me:
```
COMP-3                          Yes (aliased with COMP, COMPUTATIONAL, COMPUTATIONAL-3)
COMPUTATIONAL-3                 Yes (aliased with COMP, COMP-3, COMPUTATIONAL)
```

Reply:
The list in reserved.c is only active with `-std=default` and with the lax include (additional to the dialect specific word list).
It just looked strange to me that `COMP-5` and friends do have the alias given, while this one doesn't.

Reply:
I have added a simple test for `COMP-{9,10,15}`, but I limited the computations to a minimum as the proper keywords `FLOAT_BINARY_{32,64,128}` appear to trigger internal errors (some missing entries in `cobc/field.c (setup_parameters)`, at least). There's now an entry for that in `TODO`, in case (but I guess there's probably a pending issue about that already).